src/custom/state/claim/hooks/index.ts

```diff
@@ -180,6 +182,54 @@ export function useUserClaims(account: Account): UserClaims | null {
   return claimKey ? claimInfo[claimKey] : null
 }
 
+/**
+ * Fetches from contract the deployment timestamp in ms
+ *
+ * Returns null if in there's no network or vCowContract doesn't exist
+ */
+function useDeploymentTimestamp(): number | null {
+  const { chainId } = useActiveWeb3React()
+  const vCowContract = useVCowContract()
+  const [timestamp, setTimestamp] = useState<number | null>(null)
+
+  useEffect(() => {
+    if (!chainId || !vCowContract) {
+      return
+    }
+
+    vCowContract.deploymentTimestamp().then((ts) => {
+      console.log(`Deployment timestamp in seconds: ${ts.toString()}`)
+      setTimestamp(ts.mul('1000').toNumber())
+    })
+  }, [chainId, vCowContract])
+
+  return timestamp
+}
+
+/**
+ * Returns whether vCOW contract is still open for investments
+ * Null when not applicable
+ *
+ * That is, there has been less than 2 weeks since it was deployed
+ */
+export function useInvestmentStillAvailable(): boolean | null {
+  const deploymentTimestamp = useDeploymentTimestamp()
+
+  return deploymentTimestamp ? deploymentTimestamp + TWO_WEEKS > Date.now() : null
+}
+
+/**
+ * Returns whether vCOW contract is still open for airdrops
+ * Null when not applicable
```

Comment:
i dont get what difference between `false` and `null` are here

Reply:
`false` when we know it's closed, `null` when chainId or the contract are undefined.

Do you think we shouldn't differentiate? I don't mind updating

Reply:
Alright, removing the `null`